### PR TITLE
Make a general framework for doc in how_to_use_capi. 

### LINKED ISSUE #372: Update the documentation of "How to use CAPI" in PaddleModel repo.
- 我们计划在 models repo 的 [`how_to_use_capi`](https://github.com/PaddlePaddle/models/tree/develop/how_to_use_capi) 目录下增加更加丰富的，与CAPI使用相关的文档和例子，保证能够覆盖到 PaddlePaddle 所有输入数据类型。
- 在这个issue 中，先提出我们能够想到的，需要包含的最小信息。请大家不吝提出更好的建议，帮助我们做好“CAPI使用”相关的文档和例子。
- `how_to_use_capi` 会含有二级目录，对应不同的可以完整运行的示例代码；
- 一级目录下的 README 需要介绍与CAPI使用相关的通用信息。
- 以下是关于一级目录下 README 的一个基础提纲：
---

### 2. CAPI 使用流程指南
1.  序列化模型配置；
2. 组织数据数据；
3. 初始化/加载模型；
4. 前向计算；
5. 清理；

### 3. 从python 用户接口的输入数据类型，到 C++ 端输入数据类型对应

- 首先，需要一个`[ 12 x 2]` 的表格，第一列是用户在 python 端看到的，PaddlePaddle 支持的所有 12 种输入数据类型；第二列对应了 12 种输入数据在 C++ 代码中如何组织的tips。
- 关键代码片段和一定的解释说明；

### 5. F&Q

### PR DESCRIPTION
**Pull request by peterzhang2029**
resolve https://github.com/PaddlePaddle/models/issues/372

### CODE REVIEW

**Comment by hedaoyuan** [comment]
赞这个文档。
在Mobile上用户同样遇到`how_to_use_capi`的问题 [See Inference API](https://github.com/PaddlePaddle/Paddle/issues/5782)。另外，我有几个问题：
1. 这个PR还在开发吗？还是只差Review了？我可以来Review。
2. 这个文档可以移到Paddle Repo中吗？我觉得直接放到capi那个目录里面可能比较好。
3. 建议可以加一个图，解释整个编程模型的步骤，包括序列化模型->初始化->create gradient machine -> 组织输入数据 -> 前向计算 -> 获取数据 -> 清理。
4. 可以有独立的章节解释多线程计算，Batch Computing等。

**Comment by lcy-seso** [comment]
@hedaoyuan 这个PR会继续开发，README 我来继续写完。最近就搞定。

**Comment by wangkuiyi** [comment]
We will need English documentation. It could be in this PR, or a future PR.

For some old PRs in the PaddlePaddle/mobile repo, it was me who added the English document. It would be better if the original author can give an English draft, and I can help polish them.

**Comment by lcy-seso** [comment]
@wangkuiyi We will have a discussion with @hedaoyuan and @qingqing01 in this week to better organize this documentation and finally move it into the PaddlePaddle repo.  We will add the English version documentation into the plan.

@hedaoyuan gave us some suggestions to re-organize the documentation to make it more informative by explaining the programming model and core concepts in PaddlePaddle rather than just copying our example codes. The outline and contents of this documentation will further be re-organized in the next PR.

File: how_to_use_capi/README.md [hunk 1]
```diff
@@ -1 +1,108 @@
...
+      -DWITH_PYTHON=OFF \
+      -DWITH_MKLML=OFF \
+      -DWITH_MKLDNN=OFF \
+      -DWITH_GPU=OFF \
+      -DWITH_SWIG_PY=OFF \
+      -DWITH_GOLANG=OFF \
+make
+make install
+```
+其中，`DEST_ROOT` 表示编译得到的动态链接库和头文件的存储目录。`PADDLE_ROOT` 表示 PaddlePaddle 源码所在目录。
```
Comment (chengduoZH) [inline how_to_use_capi/README.md:20]: 这里是否可以把每个选项的意思介绍一下，比如`WITH_PYTHON=OFF`为什么要关掉，`WITH_AVX`是该打开还是该关闭等

**Pull request closed without merging**